library/core/networking/src/main/java/org/quiltmc/qsl/networking/api/codec/ArrayNetworkCodec.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.networking.api.codec;
+
+import java.util.function.Consumer;
+import java.util.function.IntFunction;
+
+import net.minecraft.network.PacketByteBuf;
+
+public final class ArrayNetworkCodec<A> implements NetworkCodec<A[]> {
+	private final NetworkCodec<A> entryCodec;
+	private final IntFunction<? extends A[]> arrayFactory;
+
+	public ArrayNetworkCodec(NetworkCodec<A> entryCodec, IntFunction<? extends A[]> arrayFactory) {
```

Comment:
There are a lot of construction methods left public through all these classes.

Are users meant to be able to construct these codecs directly, or always go through `NetworkCodec`, `NetworkCodecBuilder`, etc?